src/Test/Plutip/Internal/BotPlutusInterface/Wallet.hs

```diff
@@ -95,6 +130,15 @@ saveWallet (BpiWallet pkh _ sk) = do
       res <- liftIO $ CAPI.writeFileTextEnvelope path (Just "Payment Signing Key") key
       return $ left (SignKeySaveError . show) res --todo: better error handling
 
+-- | Save the wallet to a specific directory.
+saveWalletDir :: MonadIO m => BpiWallet -> FilePath -> m (Either BpiError ())
```

Comment:
It feels like `saveWallet` can merged with `saveWalletDir`, if we pass `Maybe FilePath` as second argument. It looks like we actually making additional copy of keys to user specified directory, coz we saving them to `Setup.keysDir` in any case.
And maybe `eitherAddSomeWallet` can be expressed via `eitherAddSomeWalletDir` then (?)

Reply:
Yeah, there are two copies of the keys. The idea was that the directory specified by the user was an additional directory that they had easier access to, and was not necessarily temporary.

Reply:
Think in terms of copying - maybe merge `save...` functions them completely?
Like make single function
```
saveWallet bpiWallet keysCopyPath  :: MonadIO m => BpiWallet -> Maybe FilePath -> m (Either BpiError ())
```
Then it will save keys to mandatory `bpi` directory (`Setup.keysDir cEnv`). And if `keysCopyPath` is `Just`, also saves there.

And in `eitherAddSomeWalletDir` instead of  
```haskell
  case wallDir of
    Nothing -> pure ()
    (Just direc) -> void $ saveWalletDir bpiWallet direc
  saveWallet bpiWallet
```
just do 
```haskell
 saveWallet bpiWallet direc
``` 
What do you think?